tests/read_accel.rs

```diff
@@ -1,25 +1,134 @@
+            sensor.set_accel_odr(AccelOutputDataRate::Hz50).unwrap();
+
+            if let AccelMode::LowPower | AccelMode::HighResolution = mode {
+                sensor.set_accel_mode(mode).unwrap();
+            }
+            if let AccelScale::Scale2g = scale {
+            } else {
+                sensor.set_accel_scale(scale).unwrap();
+            }
+
+            let actual = sensor.accel_data().unwrap();
+
+            assert_eq!(actual, expected);
+
+            destroy_i2c(sensor);
+        }
+    };
+}
+
+#[rustfmt::skip]
+mod can_get_i2c {
+    use super::*;
+
+    can_get_i2c!(low_power_2g,        LowPower,       Scale2g,  Measurement { x: 512, y: 1024, z: 1536 });
+    can_get_i2c!(high_resolution_2g,  HighResolution, Scale2g,  Measurement { x: 513, y: 1027, z: 1541 });
+    can_get_i2c!(normal_2g,           Normal,         Scale2g,  Measurement { x: 512, y: 1024, z: 1540 });
+    can_get_i2c!(low_power_4g,        LowPower,       Scale4g,  Measurement { x: 512 * 2, y: 1024 * 2, z: 1536 * 2});
+    can_get_i2c!(high_resolution_4g,  HighResolution, Scale4g,  Measurement { x: 513 * 2, y: 1027 * 2, z: 1541 * 2});
+    can_get_i2c!(normal_4g,           Normal,         Scale4g,  Measurement { x: 512 * 2, y: 1024 * 2, z: 1540 * 2});
+    can_get_i2c!(low_power_8g,        LowPower,       Scale8g,  Measurement { x: 512 * 4, y: 1024 * 4, z: 1536 * 4});
+    can_get_i2c!(high_resolution_8g,  HighResolution, Scale8g,  Measurement { x: 513 * 4, y: 1027 * 4, z: 1541 * 4});
+    can_get_i2c!(normal_8g,           Normal,         Scale8g,  Measurement { x: 512 * 4, y: 1024 * 4, z: 1540 * 4});
+    can_get_i2c!(low_power_16g,       LowPower,       Scale16g, Measurement { x: 512 * 8, y: 1024 * 8, z: 1536 * 8});
+    can_get_i2c!(high_resolution_16g, HighResolution, Scale16g, Measurement { x: 513 * 8, y: 1027 * 8, z: 1541 * 8});
+    can_get_i2c!(normal_16g,          Normal,         Scale16g, Measurement { x: 512 * 8, y: 1024 * 8, z: 1540 * 8});
+}
 
 #[test]
 fn can_get_10_bit_data() {
-    let mut sensor = new_i2c(&[I2cTrans::write_read(
-        ACCEL_ADDR,
-        vec![Register::OUT_X_L_A | 0x80],
-        vec![0x10, 0x20, 0x30, 0x40, 0x50, 0x60],
-    )]);
+    let mut sensor = new_i2c(&[
+        I2cTrans::write(
+            ACCEL_ADDR,
+            vec![Register::CTRL_REG1_A, DEFAULT_CTRL_REG1_A | HZ50],
+        ),
+        I2cTrans::write_read(
+            ACCEL_ADDR,
+            vec![Register::OUT_X_L_A | 0x80],
+            vec![0x10, 0x20, 0x30, 0x40, 0x50, 0x60],
+        ),
+    ]);
+    sensor.set_accel_odr(AccelOutputDataRate::Hz50).unwrap();
     let data = sensor.accel_data().unwrap();
     assert_eq!(
         data,
-        UnscaledMeasurement {
-            x: 0x2010 >> 6,
-            y: 0x4030 >> 6,
-            z: 0x6050 >> 6
+        Measurement {
+            x: 512,  // ~= 0x2010 / (1 << 4)
+            y: 1024, // ~= 0x4030 / (1 << 4)
+            z: 1540, // ~= 0x6050 / (1 << 4)
```

Comment:
Same?
Also below.

Reply:
these ones are slightly different because the point is they're approximately equal

Reply:
I suppose I could translate it to something like
```rust
    assert!((data.x - 0x2010 / (1 << 4)).abs() < 10)
    assert!((data.y - 0x4030 / (1 << 4)).abs() < 10)
    assert!((data.z - 0x6050 / (1 << 4)).abs() < 10)
```

Reply:
I mean, the result will be exactly the same whether the numerical operations happen inside the driver method or in the test code, right? If you feed the driver the same numbers through the transaction, the output number should be exactly the same (however off they are).
Wouldn't it also pass with just this, if that is also what happens inside the library?
```rust
x: 0x2010 / (1 << 4),
y: 0x4030 / (1 << 4),
z: 0x6050 / (1 << 4),
```
Or have you rather seen the compiler optimize the operation more accurately in the test to a constant that actually does not match because it can see the entire thing?

Reply:
What I'm trying to convey is that the result of reading `x` as `AccelMode::LowPower` and `AccelScale::Scale2g` should result in a number that is roughly equal `AccelMode::HighResolution` and `AccelScale::Scale2g`. The results will not be exactly equal as the low power one has a lower resolution.

Reply:
`0x2010 / (1 << 4)` is a convenient base case (`AccelMode::HighResolution`, `AccelScale::G2`) because it has a scaling factor of `1`.

Reply:
Ah, I understand. That makes sense. If it makes it easier to you you can also define a macro like:
```rust
macro_rules! assert_almost_eq {
    ($value:expr, $expected:expr) => {
        assert!(($value - $expected).abs() < 6)
    }
}
```

Reply:
Cool, I've done that. This conversation thread also made me realise that the intent wasn't entirely obvious so I've added some comments explaining where those numbers are coming from.